lib/src/solver/result.dart

```diff
@@ -109,6 +122,34 @@ class SolveResult {
     }
   }
 
+  /// Send analytics about the package resolution.
+  void sendAnalytics(Analytics analytics) {
+    ArgumentError.checkNotNull(analytics);
+
+    for (final package in packages) {
+      final source = package.source;
+      // Only send analytics for packages from pub.dev.
+      if (source is HostedSource &&
+          (runningFromTest ||
+              package.description['url'] == HostedSource.pubDevUrl)) {
+        final dependencyType = const {
+          DependencyType.dev: 'dev',
+          DependencyType.direct: 'direct',
+          DependencyType.none: 'transitive'
+        }[_root.dependencyType(package.name)];
+
+        analytics.sendEvent(
+          'pub-get',
+          package.name,
+          label: package.version.toString(),
+          value: 1,
+          parameters: {'kind': dependencyType},
```

Comment:
Do we know what this looks like on google analytics?

I'm sure we can always do aggregations based on this using big query regardless, but it might be worth looking into.